### PR TITLE
idle-disconnecting

### PR DESCRIPTION
**Pull request by ccassette**

### CODE REVIEW

**Comment by ccassette** [comment]
disconnects users who idle for too long

**Comment by oldmud0** [comment]
You should be able to repurpose the standard `timeout` for this.

File: config_sample/config.yaml [hunk 1]
```diff
@@ -82,3 +82,9 @@ max_chars: 256
 
 # How long a ban will last when no duration is given
 default_ban_duration: 6 hours
+
+# Kicks idlers
+idle_timeout:
+  use_idle_timeout: false
+  kick_mods: false
```
Comment (caleb-mabry) [inline config_sample/config.yaml:89]: What's the purpose of the `kick_mods` setting? I don't see you using it anywhere

File: server/tsuserver.py [hunk 2]
```diff
@@ -221,6 +229,12 @@ def load_config(self):
                 'interval_length': 0,
                 'mute_length': 0
             }
+        if 'idle_timeout' not in self.config:
+            self.config['idle_timeout'] = {
+                'use_idle_timeout': False,
+                'kick_mods': False,
+                'length': 0
```
Comment (caleb-mabry) [inline server/tsuserver.py:236]: I think that setting this to length of 0 you're allowing people to be kicked every 60 seconds if they don't have this configuration

**Review by caleb-mabry — DISMISSED** [review]
This doesn't offer a way for you to turn this setting off. By implementing this you're forcing people to use the setting.
It might also be a better idea to tie this into the AFK setting rather than trying to setup something completely separate. 

Since we are already tracking AFK users, it could probably be worthwhile to offer disconnecting people with that feature.

**Comment by Aho-Shoujo** [comment]
> This doesn't offer a way for you to turn this setting off. By implementing this you're forcing people to use the setting.

idle_timeout:
  use_idle_timeout: false


> I think that setting this to length of 0 you're allowing people to be kicked every 60 seconds if they don't have this configuration

just tried this, you did NOT try this

**Comment by oldmud0** [comment]
Reopened by a requester of this feature.

File: server/tsuserver.py (hunk 2 above)
Comment (oldmud0) [reply server/tsuserver.py:236]: yeah, this is tolerable, since these three settings will always be defined by the user if the idle timeout is enabled

**Review by oldmud0 — APPROVED** [review]
generally not much can go wrong with this.

**Pull request merged**